app_dart/lib/src/service/github_service.dart

```diff
@@ -27,4 +25,37 @@ class GithubService {
     }
     return checkRuns;
   }
+
+  /// Lists the commits of the provided repository [slug] and [branch].
+  Future<List<RepositoryCommit>> listCommits(
+      RepositorySlug slug, String branch) async {
+    const Duration time = Duration(hours: 1);
+    ArgumentError.checkNotNull(slug);
+    final PaginationHelper paginationHelper = PaginationHelper(github);
+
+    final List<dynamic> commits = <dynamic>[];
+    await for (Response response in paginationHelper.fetchStreamed(
```

Comment:
May be worth sending a PR to `package:github` to add the functionality to query commits on a branch.

Reply:
Will consider sending that soon.